build_rhel.sh

```diff
@@ -6,7 +6,13 @@
 
 # The new gcc 7 is now available using the command 'source scl_source enable devtoolset-7'
 # the previous gcc will be enabled again after this script exits
-source scl_source enable devtoolset-7
+os_name=$(grep -o '^NAME=.*' /etc/os-release | cut -f2 -d\" | sed 's/"//g')
+os_version=$(grep -o '^VERSION_ID=.*' /etc/os-release | cut -f2 -d\" | sed 's/"//g')
+# Use scl_source only if OS is RedHat/CentOS 7
+if [[ ( $os_name == *"Red Hat"* || $os_name == *"CentOS"* ) &&  $os_version == *"7"* ]]
+then
+    source scl_source enable devtoolset-7
+fi
```

Comment:
build_rhel.sh is called when the OS is RedHat or CentOS, so why did we check the OS name and version again in this script?

Reply:
I'll remove the check for OS name, but we are checking OS _VERSION because scl_source works in Centos 7 only not in Centos Stream 9.
Its support is available till RHEL/CentOS 7 only.